### PR TITLE
BRS-170: park search improvements

### PR DESCRIPTION
**Pull request by cole**
# Description

This PR includes the following proposed change(s):

Removes the search/search-view parts of strapi to try (with kind of mixed success tbh) to replicate some of the default strapi search behaviour (same params, etc). Keyword search is done using Postgres full text search with added indexes.

The changes to the explore page are to keep everything working, also removes the initial (static) load of protected areas.

Unfortunately the API side makes heavy use of raw queries via knex, which should protect from SQL injection but ties us to using Postgres.

Note that some searches are quite slow (e.g. 1.5s query time), possibly due to missing indexes with a large number of joins. Needs a bit of investigation, but deferring that for now.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

### CODE REVIEW

File: src/cms/data/functions/createSearchIndexes.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+module.exports = async () => {
+  try {
+    const knex = strapi.connections[strapi.config.database.defaultConnection];
+
+    const hasSearchColumn = await knex.schema.hasColumn(
+      "protected_areas",
+      "search_text"
+    );
+
+    if (!hasSearchColumn) {
```
Comment (cole) [inline src/cms/data/functions/createSearchIndexes.js:13]: @BcGovNeal this seemed like a reasonably safe way to handle index & generated column creation. Let me know if you have concerns or can see ways to improve it.

Comment (BcGovNeal) [reply src/cms/data/functions/createSearchIndexes.js:13]: I think this is fine as long as we don't introduce other ways of modifying the same schema

**Review by BcGovNeal — COMMENTED** [review]
I'm noticing the search query is fairly slow.  I'm getting any where between 1s to 1.7s on my machine.  This might be okay for beta but we'll probably need to improve on this eventually.

Since it takes a while to get the search results back, we probably should add some sort of loading indicator

File: src/staging/src/pages/explore.js [hunk 2]
```diff
@@ -333,108 +281,123 @@ export default function Explore({ location, data }) {
...
-    protectedAreas,
     data.site.siteMetadata.apiURL,
-    processResults,
     setFilters,
-    isFilteredResult,
-    filteredResults,
+    setNumberOfPages,
+    setSearchResults,
+    setTotalResults,
+    totalResults,
```
[the remark this answers is not in the document]
Comment (cole) [reply src/staging/src/pages/explore.js:400]: Ah, oops, thank you! Will remove

**Comment by cole** [comment]
> I'm noticing the search query is fairly slow. I'm getting any where between 1s to 1.7s on my machine. This might be okay for beta but we'll probably need to improve on this eventually.

I added https://bcparksdigital.atlassian.net/browse/BRS-334 but we may just want to make that to improve performance.

**Review by DayKirby-OXD — APPROVED** [review]
Thanks for adding inline doc'n!

File: src/staging/src/pages/explore.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cole) [reply src/staging/src/pages/explore.js]: fixed in 8c41502

**Comment by cole** [comment]
Added loading indicator in 02a08dd and updated benchmark ticket to be about improving performance.

**Review by BcGovNeal — APPROVED** [review]
Looks good!  I like the loading indicator!

**Pull request merged**